exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1188 @@
+	for x := range withdrawals {
+		resp = append(resp, exchange.WithdrawalHistory{
+			Status:          withdrawals[x].StateOfWithdrawal,
+			Timestamp:       withdrawals[x].Timestamp,
+			Currency:        withdrawals[x].Currency,
+			Amount:          withdrawals[x].Amount,
+			TransferType:    "withdrawal",
+			CryptoToAddress: withdrawals[x].ToReceivingAddress,
+			CryptoTxID:      withdrawals[x].TransactionID,
+			CryptoChain:     withdrawals[x].ChainName,
+			TransferID:      withdrawals[x].WithdrawalID,
+			Fee:             withdrawals[x].WithdrawalFee,
+		})
+	}
+	return resp, nil
+}
+
+// GetRecentTrades returns the most recent trades for a currency and asset
+func (ok *Okx) GetRecentTrades(ctx context.Context, p currency.Pair, assetType asset.Item) ([]trade.Data, error) {
+	instrumentID, err := ok.getInstrumentIDFromPair(ctx, p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	tradeData, err := ok.GetTrades(ctx, instrumentID, 1000)
+	if err != nil {
+		return nil, err
+	}
+	resp := make([]trade.Data, len(tradeData))
+	for x := range tradeData {
+		resp[x] = trade.Data{
+			TID:          tradeData[x].TradeID,
+			Exchange:     ok.Name,
+			CurrencyPair: p,
+			AssetType:    assetType,
+			Price:        tradeData[x].Price,
+			Amount:       tradeData[x].Quantity,
+			Timestamp:    tradeData[x].Timestamp,
+		}
+	}
+	if ok.IsSaveTradeDataEnabled() {
+		err := trade.AddTradesToBuffer(ok.Name, resp...)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sort.Sort(trade.ByDate(resp))
+	return resp, nil
+}
+
+// GetHistoricTrades returns historic trade data within the timeframe provided
+func (ok *Okx) GetHistoricTrades(ctx context.Context, p currency.Pair, assetType asset.Item, timestampStart, timestampEnd time.Time) ([]trade.Data, error) {
+	const limit = 1000
+	instrumentID, err := ok.getInstrumentIDFromPair(ctx, p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	tradeData, err := ok.GetTradesHistory(ctx, instrumentID, "", "", limit)
+	if err != nil {
+		return nil, err
+	}
+	resp := make([]trade.Data, len(tradeData))
+	for x := range tradeData {
+		resp[x] = trade.Data{
+			TID:          tradeData[x].TradeID,
+			Exchange:     ok.Name,
+			CurrencyPair: p,
+			AssetType:    assetType,
+			Price:        tradeData[x].Price,
+			Amount:       tradeData[x].Quantity,
+			Timestamp:    tradeData[x].Timestamp,
+		}
+	}
+	if ok.IsSaveTradeDataEnabled() {
+		err := trade.AddTradesToBuffer(ok.Name, resp...)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sort.Sort(trade.ByDate(resp))
+	return resp, nil
+}
+
+// SubmitOrder submits a new order
+func (ok *Okx) SubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	var submitOrderResponse order.SubmitResponse
+	if err := s.Validate(); err != nil {
+		return nil, err
+	}
+	orderType, err := ok.OrderTypeString(s.Type)
+	if err != nil && !(s.AssetType == asset.PerpetualSwap || s.AssetType == asset.Futures) {
+		return nil, errInvalidOrderType
+	}
```

Comment:
I think it will need to be rewritten

If there is an error on your defined function, and its a perpetual swap, allow any order type? The docs do not suggest this: https://www.okx.com/docs/en/#rest-api-trade-place-order

Reply:
there is no equivalent field for the trade mode in the order. Submit. So, should I add my own field to It?
If not, it will still be confusing.
Let me hear your suggestion after the change and make change accordingly